app/services/course/assessment/reminder_service.rb

```diff
@@ -10,9 +10,40 @@ def opening_reminder(user, assessment, token)
     Course::AssessmentNotifier.assessment_opening(user, assessment)
   end
 
-  def closing_reminder(user, assessment, token)
+  def closing_reminder(assessment, token)
     return unless assessment.closing_reminder_token == token && assessment.published?
 
-    Course::AssessmentNotifier.assessment_closing(user, assessment)
+    # Send reminder emails to each student who hasn't submitted.
+    recipients = uncompleted_students(assessment)
+    recipients.each do |recipient|
+      Course::Mailer.assessment_closing_reminder_email(assessment, recipient).deliver_later
+    end
+
+    # Send an email to each instructor with a list of students who haven't submitted.
+    course_instructors = assessment.course.instructors.map(&:user)
+    course_instructors.each do |instructor|
+      Course::Mailer.assessment_closing_summary_email(
+        instructor, assessment, uncompleted_students_list(recipients)
+      ).deliver_later
+    end
+  end
+
+  private
+
+  def uncompleted_students(assessment)
+    course_users = assessment.course.course_users
+    students = course_users.student.includes(:user).map(&:user)
```

Comment:
actually `course.users.merge(CourseUser.student)` is faster, no need to instantiate intermediate records (course_users).